### PR TITLE
Fix support for plain LightningModules with FlashDataModule

### PR DESCRIPTION
**Pull request by krshrimali**
## What does this PR do?

Before this PR, there would be an error of missing attribute `process_<train/val/test/predict>_dataset` while using a Flash `DataModule` with plain `LightningModule`.

## Before submitting
- [ ] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the **[contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md)**, Pull Request section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes?
- [x] Did you write any **new necessary tests**? [not needed for typos/docs]
- [x] Did you verify **new and existing tests pass** locally with your changes?
- [ ] If you made a notable change (that affects users), did you **update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)**?

## PR review
 - [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/core/data/test_data_module.py [hunk 1]
```diff
@@ -139,6 +140,19 @@ def on_fit_end(self) -> None:
...
+            super().__init__()
+            self.layer = torch.nn.Linear(2, 1)
+
+    model = SampleBoringModel()
+    trainer = Trainer(fast_dev_run=True)
+    trainer.fit(model, datamodule=dm)
+    trainer.validate(model, datamodule=dm)
+    trainer.test(model, datamodule=dm)
+    trainer.predict(model, datamodule=dm)
+
```
Comment (krshrimali) [inline tests/core/data/test_data_module.py:155]: I think this test file in its current state, can be updated. There is a lot that can be better-engineered, but let's keep it for another PR! :)

**Comment by krshrimali** [comment]
@ethanwharris - Regarding the doc update, can you help link the file/section that should be revised that it's now supported?

**Review by Borda — APPROVED** [review]
lgtm

**Review by ethanwharris — APPROVED** [review]
